### PR TITLE
Ensure batch uses passed eauth token or credentials

### PR DESCRIPTION
**Pull request by mchugh19**
### What does this PR do?
Pass eauth credentials to the batch client.

### What issues does this PR fix or reference?
Batch support was removed from saltapi due to not honoring eauth in #38497

### Previous Behavior
Before batch support was removed from saltapi, eauth creds like
```
external_auth:
  pam:
    jenkins:
      - state.*
      - '@runner'
```
Would still allow running a cmd.run in batch mode

### New Behavior
Now saltapi returns 401 Unauthorized as expected

### Tests written?
No

@whiteinge Please test and review. This seems to do the trick for me, but as it involves enabling a potential security hole, I'd like lots of eyes on it.

### CODE REVIEW

**Comment by cachedout** [comment]
@whiteinge Could you swing by here when you get a moment please?

File: salt/client/__init__.py
Comment (whiteinge) [inline salt/client/__init__.py]: Very close! `kwargs` here may contain `token`, `username`, `password`, and `eauth`. That last one is the name of the backend (e.g., `pam`) rather than the dictionary of those values that the `Batch` class is expecting.

**Review by whiteinge — CHANGES_REQUESTED** [review]
Sorry for the slow reply here. Very cool! Thanks so much for diving in to get this re-added.

One small change, I'll give this another test, then I think this is ready.

File: salt/client/__init__.py
Comment (whiteinge) [inline salt/client/__init__.py]: Er, my comment wasn't very clear. It will be called like this:

```
local = salt.client.LocalClient()
local.cmd_batch('*', 'test.ping', username='saltdev', password='saltdev', eauth='auto')
```

Comment (mchugh19) [inline salt/client/__init__.py]: Just updated to get all of `eauth`, `username`, `password`, and `token` to pass into the batch class. Is this the desired behavior?

Comment (whiteinge) [inline salt/client/__init__.py]: Almost. You'll need `kwargs['eauth']` too.

One other thought: those four values should be pop'ed out of kwargs so that we don't accidentally send them to minions.

Comment (mchugh19) [inline salt/client/__init__.py]: Third try's the charm?

Comment (whiteinge) [inline salt/client/__init__.py]: Thanks. We're nearly there! Those values won't always be present so pop'ing without a default value will throw a `KeyError`.

If you want to try this at the Python REPL, typing the lines below should do the trick. The three scenarios we care about are: eauth with creds, eauth with an existing token, no eauth.

```py
import salt.client
local = salt.client.LocalClient()
local.cmd_batch('*', 'test.ping', username='<username>', password='<password>', eauth='<eauth backend>')
local.cmd_batch('*', 'test.ping', token='<eauth token>')
local.cmd_batch('*', 'test.ping')
```

Comment (whiteinge) [inline salt/client/__init__.py]: Oh, and since that method returns a generator you can view the results via:

```py
ret = local.cmd_batch('*', 'test.ping')
for i in ret: print i
```

Comment (mchugh19) [inline salt/client/__init__.py]: I don't think it's pop'ing without a default. The pops are wrapped by `if 'username' in kwargs:` statements. Is that not sufficient?

Comment (whiteinge) [inline salt/client/__init__.py]: I apologize! I've been doing a bit too much context switching this morning which has apparently broken my ability to read good. Pulling this down locally to test now...

**Review by whiteinge — APPROVED** [review]

**Comment by whiteinge** [comment]
@mchugh19 thank you for seeing this through and getting this fixed!

It looks like we're in good shape to merge this. If you would like to restore the removal from `salt/netapi/__init__.py` in 3d8f3d18f6af as part of this PR I think that is very appropriate and will get anything using that class batch support, like `rest_cherrypy`. (Restoring Saltnado will need to be a separate thing.)

**Comment by whiteinge** [comment]
Switched the target branch over to Nitrogen so we can get this in the next release.

**Comment by mchugh19** [comment]
Cool cool. Re-added batch support to cherrypy. Could/should this be backported to 2016.11.04 or 05? Since it was released with batch support that was later removed, it seems appropriate to fix.

**Comment by whiteinge** [comment]
That's a fair argument. I'll defer to the Core team. If we decide to go that route we could rebase this there and then merge-forward it as normal for Nitrogen.

**Comment by cachedout** [comment]
That argument is completely fair. I will mark this as a backport. Thanks, @mchugh19

**Pull request merged**
